### DISCUSSION TITLE
How to perform preprocessing using Adaptive Scaler

### DISCUSSION THREAD

**Discussion by medha-chippa** [online-ml/river#690] [Q&A] [answered]
Post the prediction task, one of the predictions turns out to be 'None'. Is this because of insufficient data or because of incorrect preprocessing. Note that this was just a small piece of reproducible code, as result of which the same preprocessing method has been used twice. One of the preprocessing methods shall later be modified to minmaxscaler or other similar methods. The following is the piece of code:

students=pd.DataFrame()

students['r_no']=[1,2,3,4,5]
students['course_no']=['CS_123','CS_456','CS_890','CS_678','CS_123']
students['pass']=[1,1,0,0,1]

x_students=students.drop(['pass'],axis=1)
y_students=students['pass']

strm_st1=iter_pandas(X=x_students,y=y_students)
strm_st2=iter_pandas(X=x_students,y=y_students)

metric1 = metrics.Accuracy()
metric2 = metrics.Accuracy()

y_preds_st1 = []
y_preds_st2 = []

for ((xi1, yi1),(xi2, yi2)) in zip(strm_st1,strm_st2):
  trans_xi1=scaler1.learn_one(xi1).transform_one(xi1)
  trans_xi2=scaler2.learn_one(xi2).transform_one(xi2)
  pred_i1=c1.predict_one(trans_xi1)
  pred_i2=c2.predict_one(trans_xi2)
  c1.learn_one(trans_xi1, yi1)

  c2.learn_one(trans_xi2, yi2)
  metric1.update(y_pred=pred_i1, y_true=yi1)

  metric2.update(y_pred=pred_i2, y_true=yi2)
  y_preds_st1.append(pred_i1)

  y_preds_st2.append(pred_i2)

print(f"Accuracy for c1: {metric1.get()}")

print(f"Accuracy for c2: {metric2.get()}")

Both y_preds_st1 and y_preds_st2 yield [None,1,1,1,1]

**Comment by MaxHalford** [comment, marked as answer]
What do you mean by "Adaptive Scaler"? `preprocessing.AdaptiveStandardScaler`?

I'm not sure what you're asking for. We're not here to code stuff for you. We're here to help you with troubles in your code. Can you tell us what you have tried? Have you even taken a look at the documentation?

Meta: please make an effort at formatting your code and making it executable. It's tiring to have to have to read through discussion posts where very little effort is put into formatting.

**Reply by medha-chippa** [reply]
I apologize for the lack of sufficient context. Yes, I am referring to preprocessing.AdaptiveStandardScaler. The code has been modified, please have a look.

Thank you

**Reply by MaxHalford** [reply]
The code is not runnable. For instance what's `c1` and `c2`? What about `scaler1` and `scaler2`? Please provide a [minimal reproducible example](https://stackoverflow.com/help/minimal-reproducible-example). Also, what is your question? Do you have a bug? Some error message you don't understand?
